pkg/router/template/util/haproxy/config_parser.go

```diff
@@ -0,0 +1,165 @@
+package haproxy
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"strings"
+)
+
+const (
+	commentChar     = "#"
+	globalKeyword   = "global"
+	defaultsKeyword = "defaults"
+	frontendKeyword = "frontend"
+	backendKeyword  = "backend"
+	edgeBePrefix    = "be_edge_http"
+)
+
+// ConfigParser is a primitive HAProxy config parser.
```

Comment:
I'm wondering whether you looked at https://github.com/haproxytech/config-parser and if that would do what we need.

Reply:
No, I didn't, I wasn't expecting any golang libs from HAProxy community. But this one is even from the upstream! Thanks a lot for the advice. This module looks very promising:
```golang
$ cat config.cfg  | grep -A 15 be_edge_http:test-route:hello-openshift2
backend be_edge_http:test-route:hello-openshift2
  mode http
  option redispatch
  option forwardfor
  balance 

  timeout check 5000ms
  http-request add-header X-Forwarded-Host %[req.hdr(host)]
  http-request add-header X-Forwarded-Port %[dst_port]
  http-request add-header X-Forwarded-Proto http if !{ ssl_fc }
  http-request add-header X-Forwarded-Proto https if { ssl_fc }
  http-request add-header X-Forwarded-Proto-Version h2 if { ssl_fc_alpn -i h2 }
  http-request add-header Forwarded for=%[src];host=%[req.hdr(host)];proto=%[req.hdr(X-Forwarded-Proto)]
  http-response set-header Strict-Transport-Security 'max-age=99999;includesubdomain'
  cookie f7e936ebe97cd64e8c888d1eba08cb29 insert indirect nocache httponly secure attr SameSite=None
  server pod:hello-openshift-7b8c68587c-mtck6:hello-openshift:8080-tcp:10.217.0.24:8080 10.217.0.24:8080 cookie 97ddca1460eca46376928fd7bcf8c89c weight 256 check inter 5000ms

$ cat main.go 
package main

import (
	"fmt"
	"github.com/haproxytech/config-parser/v4"
	"github.com/haproxytech/config-parser/v4/options"
	"github.com/haproxytech/config-parser/v4/types"
)

func main() {
	p, err := parser.New(options.Path("config.cfg"))
	if err != nil {
		fmt.Println("failed to parse the config:", err)
	}

	data, err := p.Get(parser.Backends, "be_edge_http:test-route:hello-openshift2", "http-response", false)
	if err != nil {
		fmt.Println("failed to get backend:", err)
	}

	responses := data.([]types.HTTPAction)
	for _, r := range responses {
		fmt.Println(r)
	}
}

$ go run main.go 
set-header Strict-Transport-Security 'max-age=99999;includesubdomain'
```

Let me try to integrate it into my PR.